lib/libvfio-user.c

```diff
@@ -1052,6 +1052,45 @@ handle_dirty_pages(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
     return ret;
 }
 
+static int
+handle_device_feature(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
+{
+    struct vfio_user_device_feature *req = msg->in.iov.iov_base;
+
+    if (vfu_ctx->migration == NULL) {
```

Comment:
Is this necessary? Won't `migration_feature_supported` fail anyway?

Reply:
At the moment migration is all-or-nothing and `migration_feature_supported` assumes we support migration... I don't think it makes sense to support another option until we also implement DMA logging (which will definitely be optional). What do you think?

Reply:
I feel like it makes more sense to call it `migration_feature_flags` and return the supported flags for the given feature, then have the check for migration unsupported elsewhere?

Reply:
makes sense